cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -506,32 +510,34 @@ public ClusterModel clusterModel(long from,
     final Timer.Context ctx = _clusterModelCreationTimer.time();
     try {
       // Create the racks and brokers.
-      // Shuffle nodes before getting their capacity from the capacity resolver.
-      // This enables a capacity resolver to estimate the capacity of the nodes, for which the capacity retrieval has
-      // failed.
+      // If broker capacity is allowed to estimate broker capacity, shuffle nodes before getting their capacity from the
+      // capacity resolver. This is good for the capacity resolver to estimate the capacity of the nodes, for which the
+      // capacity retrieval has failed.
       // The use case for this estimation is that if the capacity of one of the nodes is not available (e.g. due to some
       // 3rd party service issue), the capacity resolver may want to use the capacity of a peer node as the capacity for
       // that node.
       // To this end, Cruise Control handles the case that the first node is problematic so the capacity resolver does
       // not have the chance to get the capacity for the other nodes.
       // Shuffling the node order helps, as the problematic node is unlikely to always be the first node in the list.
       List<Node> shuffledNodes = new ArrayList<>(cluster.nodes());
```

Comment:
Do we even need to create a new `ArrayList` in case `allowCapacityEstimation` is false?